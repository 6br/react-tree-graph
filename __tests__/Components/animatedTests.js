import React from 'react';
import { mount, shallow } from 'enzyme';

import { easeQuadOut } from 'd3-ease';
import Animated from '../../Components/animated';

jest.useFakeTimers();

class DummyComponent extends React.PureComponent{
	render() {
		return <div/>;
	}
}

describe('<Animated>', () => {
	test('renders correctly and sets initial state', () => {
		const props = {
			animatedProps: [
				{
					name: 'x'
				}, {
					name: 'y'
				}
			],
			component: DummyComponent,
			duration: 1,
			easing: easeQuadOut,
			enabled: true,
			steps: 1,
			x: 2,
			y: 3
		};

		const tree = shallow(<Animated {...props}/>);
		expect(tree).toMatchSnapshot();

		expect(tree.state()).toMatchObject({
			x: 2,
			y: 3
		});
	});

	test('animates when props change', () => {
		const props = {
			animatedProps: [
				{
					name: 'x'
				}, {
					name: 'y'
				}
			],
			component: DummyComponent,
			duration: 100,
			easing: easeQuadOut,
			enabled: true,
			steps: 2,
			x: 2,
			y: 3
		};

		const tree = shallow(<Animated {...props}/>);
		
		tree.setProps({x: 5, y: 6});

		jest.runTimersToTime(50);

		expect(tree.state()).toMatchObject({
			x: 4.25,
			y: 5.25
		});
		expect(tree).toMatchSnapshot();

		jest.runTimersToTime(50);

		expect(tree.state()).toMatchObject({
			x: 5,
			y: 6
		});
		expect(tree).toMatchSnapshot();
	});

	test('animates from inital value on mount', () => {
		const props = {
			animatedProps: [
				{
					name: 'x',
					initialValue: 1
				}, {
					name: 'y',
					initialValue: 1
				}
			],
			component: DummyComponent,
			duration: 100,
			easing: easeQuadOut,
			enabled: true,
			steps: 1,
			x: 2,
			y: 3
		};

		const tree = mount(<Animated {...props}/>);
		
		expect(tree.state()).toMatchObject({
			x: 1,
			y: 1
		});

		jest.runTimersToTime(100);

		expect(tree.state()).toMatchObject({
			x: 2,
			y: 3
		});
	});

	test('does nothing when not enabled', () => {
		const props = {
			animatedProps: [
				{
					name: 'x'
				}, {
					name: 'y'
				}
			],
			component: DummyComponent,
			duration: 1,
			easing: easeQuadOut,
			enabled: false,
			steps: 1,
			x: 2,
			y: 3
		};

		const tree = shallow(<Animated {...props}/>);
		
		tree.setProps({x: 5, y: 6});

		expect(tree).toMatchSnapshot();
	});

	test('does nothing when props change does not change animated value', () => {
		const props = {
			animatedProps: [
				{
					name: 'x'
				}, {
					name: 'y'
				}
			],
			component: DummyComponent,
			duration: 100,
			easing: easeQuadOut,
			enabled: true,
			steps: 2,
			x: 2,
			y: 3,
			z: 4
		};

		const tree = shallow(<Animated {...props}/>);
		
		tree.setProps({z: 7});

		expect(tree.state()).toMatchObject({
			x: 2,
			y: 3
		});
	});
});