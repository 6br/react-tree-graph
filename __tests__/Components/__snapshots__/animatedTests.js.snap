// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Animated> animates when props change 1`] = `
<DummyComponent
  animatedProps={
    Array [
      Object {
        "name": "x",
      },
      Object {
        "name": "y",
      },
    ]
  }
  component={[Function]}
  duration={100}
  enabled={true}
  steps={2}
  x={2}
  y={3}
/>
`;

exports[`<Animated> animates when props change 2`] = `
<DummyComponent
  animatedProps={
    Array [
      Object {
        "name": "x",
      },
      Object {
        "name": "y",
      },
    ]
  }
  component={[Function]}
  duration={100}
  enabled={true}
  steps={2}
  x={2}
  y={3}
/>
`;

exports[`<Animated> does nothing when not enabled 1`] = `
<DummyComponent
  animatedProps={
    Array [
      Object {
        "name": "x",
      },
      Object {
        "name": "y",
      },
    ]
  }
  component={[Function]}
  duration={1}
  enabled={false}
  steps={1}
  x={5}
  y={6}
/>
`;

exports[`<Animated> renders correctly and sets initial state 1`] = `
<DummyComponent
  animatedProps={
    Array [
      Object {
        "name": "x",
      },
      Object {
        "name": "y",
      },
    ]
  }
  component={[Function]}
  duration={1}
  enabled={true}
  steps={1}
  x={2}
  y={3}
/>
`;
